larpandoracontent/LArMonitoring/TwoViewTransverseTracksValidationTool.cc

```diff
@@ -0,0 +1,164 @@
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool TwoViewTransverseTracksValidationTool::Run(const Cluster *const pCluster1, const Cluster *const pCluster2,
+    const DiscreteProbabilityVector &discreteProbabilityVector1, const DiscreteProbabilityVector &discreteProbabilityVector2,
+    const TwoViewTransverseOverlapResult &overlapResult)
+{
+
+    TreeDataBox treeDataBox;
+
+    this->CollectTruthInformation(pCluster1, treeDataBox, "1");
+    this->CollectTruthInformation(pCluster2, treeDataBox, "2");
+
+    StoreAndRegisterDatum(static_cast<int>(this->IsSameMCParticle(pCluster1, pCluster2)), "isSameMCParticle", treeDataBox);
+
+    StoreAndRegisterDatum(static_cast<int>(discreteProbabilityVector1.GetSize()), "discProbVecSize1", treeDataBox);
+    StoreAndRegisterDatum(static_cast<int>(discreteProbabilityVector2.GetSize()), "discProbVecSize2", treeDataBox);
+    StoreAndRegisterDatum(static_cast<int>(overlapResult.GetNSamplingPoints()), "nSamplingPts", treeDataBox);
+    StoreAndRegisterDatum(static_cast<int>(overlapResult.GetNMatchedSamplingPoints()), "nMatchedSamplingPts", treeDataBox);
+    StoreAndRegisterDatum(static_cast<int>(overlapResult.GetNReUpsampledSamplingPoints()), "nReUpsampSamplingPts", treeDataBox);
+    StoreAndRegisterDatum(static_cast<int>(overlapResult.GetNMatchedReUpsampledSamplingPoints()), "nMatchedReUpsampSamplingPts", treeDataBox);
+    StoreAndRegisterDatum(overlapResult.GetCorrelationCoefficient(), "correlationCoef", treeDataBox);
+    StoreAndRegisterDatum(overlapResult.GetLocallyMatchedFraction(), "locallyMatchedFrac", treeDataBox);
+    StoreAndRegisterDatum(overlapResult.GetMatchingScore(), "matchingScore", treeDataBox);
+    StoreAndRegisterDatum(overlapResult.GetTwoViewXOverlap().GetTwoViewXOverlapSpan(), "xOverlapSpan", treeDataBox);
+    StoreAndRegisterDatum(overlapResult.GetTwoViewXOverlap().GetXOverlapFraction0(), "xOverlapFrac1", treeDataBox);
+    StoreAndRegisterDatum(overlapResult.GetTwoViewXOverlap().GetXOverlapFraction1(), "xOverlapFrac2", treeDataBox);
+
+    CollectChargeProfileInformation(discreteProbabilityVector1, treeDataBox, "1");
+    CollectChargeProfileInformation(discreteProbabilityVector2, treeDataBox, "2");
+
+    PANDORA_MONITORING_API(FillTree(this->GetPandora(), m_treeName.c_str()));
+
+    return true;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool TwoViewTransverseTracksValidationTool::IsSameMCParticle(const Cluster *const pCluster1, const Cluster *const pCluster2)
+{
+    const MCParticle *const pMCParticle1(MCParticleHelper::GetMainMCParticle(pCluster1));
+    const MCParticle *const pMCParticle2(MCParticleHelper::GetMainMCParticle(pCluster2));
+
+    if (pMCParticle1 && pMCParticle2)
+    {
+        return pMCParticle1->GetUid() == pMCParticle2->GetUid();
+    }
```

Comment:
I don't really mind this, but style is to remove braces here?

Reply:
Got the braces.  TY @lhwhitehead